worm_war/sources/board.cpp

```diff
@@ -15,9 +19,21 @@ void Board::addWorm(WormType type, int x, int y) {
   // updated as well to have full information about the worm that is being
   // added. They are usually based on worm's id which should be uniquely given
   // here.
+    wormTypes_ [nextId_] = type;
+    board_[x][y] = nextId_;
+    if (type == Lazy) {
+        worms_[nextId_] = std::thread(LazyWorm(x,y,this),nextId_);
```

Comment:
Be aware that this line creates a thread, runs it and then creates a new thread which is inserted into the map and joined with the first one. It works correctly, but not exactly how we'd think of it. (assignment operator (operator=) is called)